### PR TITLE
Invoice validation #87 

### PR DESCRIPTION
**Pull request by Aleksalje**
Invoice validation #87

### CODE REVIEW

File: src/components/invoices/Dialog.vue [hunk 1]
```diff
@@ -65,7 +61,9 @@
             type="number"
             label="Amount"
             lazy-rules
-            :rules="[(val) => val > 0 || 'Amount is must be bigger than zero']"
+            :rules="[
+              (val) => isValidAmount || 'Amount is must be bigger than zero',
```
Comment (MilosPaunovic) [inline src/components/invoices/Dialog.vue:65]: Check copy for error message.

File: src/components/invoices/Dialog.vue [hunk 2]
```diff
@@ -83,17 +81,19 @@
             type="number"
             label="Exchange rate"
             lazy-rules
-            :rules="[(val) => val > 0 || 'Amount is must be bigger than zero']"
+            :rules="[
+              (val) => isValidExchange || 'Amount is must be bigger than zero',
```
Comment (MilosPaunovic) [inline src/components/invoices/Dialog.vue:85]: Check copy for error message.

**Review by MilosPaunovic — CHANGES_REQUESTED** [review]
Nice direction, @Aleksalje!

I just think its better, performance wise, to wrap inputs in a`QForm` component than to create separate computed properties for every field.

That way you could revert rules to be field specific, you could leave the `Add` button always enabled but inside a method that handles adding or editing, to validate form first, and in case of an error, mark invalid fields and focus the first errored input from the top.

**Comment by MilosPaunovic** [comment]
@Aleksalje @jelenamilanovic 

Take a look at this [document](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) when you find the time, specifically, the PR keywords which will help us automate closing issues.

File: src/components/invoices/Dialog.vue (hunk 1 above)
Comment (Aleksalje) [reply src/components/invoices/Dialog.vue]: Copy corrected

File: src/components/invoices/Dialog.vue (hunk 2 above)
Comment (Aleksalje) [reply src/components/invoices/Dialog.vue]: Copy corrected

**Comment by Aleksalje** [comment]
Reverted computed rules for fields. 
The issue was that the button was emitting an event to close the popup.
Removed close-popup event and changed the button type to submit.

**Pull request merged**